src/xcode/ENA/ENA/Source/Services/Event Registration/TraceWarningDownload/__tests__/TraceWarningPackageDownloadTests.swift

```diff
@@ -0,0 +1,642 @@
+			client: ClientMock(),
+			store: MockTestStore(),
+			eventStore: eventStore
+		)
+	
+		XCTAssertEqual(eventStore.checkinsPublisher.value.count, 3)
+		
+		// WHEN
+		let earliestPackage = traceWarningPackageDownload.earliestRelevantPackage
+		
+		// THEN
+		XCTAssertEqual(earliestPackage, earliestPackageId)
+	}
+	
+	func testGIVEN_TraceWarningDownload_WHEN_CleanUpOutdatedMetadata_THEN_IsRevoked() {
+		
+		// GIVEN
+		let eventStore = MockEventStore()
+		
+		let earliestPackageId = 44440
+		let earliestPackageIdPlusOne = 44441
+		let earliestPackageIdPlusTwo = 44442
+		let earliestPackageIdPlusThree = 44443
+		let earliestPackageIdPlusFour = 44444
+
+		eventStore.createTraceWarningPackageMetadata(TraceWarningPackageMetadata(id: earliestPackageId, region: "DE", eTag: "FakeETag"))
+		eventStore.createTraceWarningPackageMetadata(TraceWarningPackageMetadata(id: earliestPackageIdPlusOne, region: "DE", eTag: "FakeETag"))
+		eventStore.createTraceWarningPackageMetadata(TraceWarningPackageMetadata(id: earliestPackageIdPlusTwo, region: "DE", eTag: "FakeETag"))
+		eventStore.createTraceWarningPackageMetadata(TraceWarningPackageMetadata(id: earliestPackageIdPlusThree, region: "DE", eTag: "FakeETag"))
+		eventStore.createTraceWarningPackageMetadata(TraceWarningPackageMetadata(id: earliestPackageIdPlusFour, region: "DE", eTag: "FakeETag"))
+		
+		let traceWarningPackageDownload = TraceWarningPackageDownload(
+			client: ClientMock(),
+			store: MockTestStore(),
+			eventStore: eventStore
+		)
+	
+		XCTAssertEqual(eventStore.traceWarningPackageMetadatasPublisher.value.count, 5)
+		
+		// WHEN
+		traceWarningPackageDownload.cleanUpOutdatedMetadata(oldest: earliestPackageId, earliest: earliestPackageIdPlusThree)
+		
+		// THEN
+		XCTAssertEqual(eventStore.traceWarningPackageMetadatasPublisher.value.count, 2)
+		XCTAssertFalse(eventStore.traceWarningPackageMetadatasPublisher.value.contains(where: { $0.id == earliestPackageId }))
+		XCTAssertFalse(eventStore.traceWarningPackageMetadatasPublisher.value.contains(where: { $0.id == earliestPackageIdPlusOne }))
+		XCTAssertFalse(eventStore.traceWarningPackageMetadatasPublisher.value.contains(where: { $0.id == earliestPackageIdPlusTwo }))
+		XCTAssertTrue(eventStore.traceWarningPackageMetadatasPublisher.value.contains(where: { $0.id == earliestPackageIdPlusThree }))
+		XCTAssertTrue(eventStore.traceWarningPackageMetadatasPublisher.value.contains(where: { $0.id == earliestPackageIdPlusFour }))
+	}
+	
+	func testGIVEN_TraceWarningDownload_WHEN_DeterminePackagesToDownload_THEN_PackagesNotInDatabaseIsReturned() {
+		
+		// GIVEN
+		let eventStore = MockEventStore()
+		let earliestPackageId = 44440
+		let earliestPackageIdPlusOne = 44441
+		let earliestPackageIdPlusTwo = 44442
+		let earliestPackageIdPlusThree = 44443
+		let earliestPackageIdPlusFour = 44444
+
+		eventStore.createTraceWarningPackageMetadata(TraceWarningPackageMetadata(id: earliestPackageId, region: "DE", eTag: "FakeETag"))
```

Comment:
Regarding another comment above. Could you please test this: remove this line and add earliestPackageId to availables?

Reply:
Test works as it should. See comment above.